GLMakie/src/screen.jl

```diff
@@ -3,13 +3,141 @@ const ZIndex = Int
 # ID, Area, clear, is visible, background color
 const ScreenArea = Tuple{ScreenID, Scene}
 
-abstract type GLScreen <: AbstractScreen end
+function renderloop end
 
-mutable struct Screen <: GLScreen
-    glscreen::GLFW.Window
+"""
+## Renderloop
+
+* `renderloop = GLMakie.renderloop`: sets a function `renderloop(::GLMakie.Screen)` which starts a renderloop for the screen.
+
+
+    !!! warning
+    The below are not effective if renderloop isn't set to `GLMakie.renderloop`, unless implemented in custom renderloop:
+
+
+* `pause_renderloop = false`: creates a screen with paused renderlooop. Can be started with `GLMakie.start_renderloop!(screen)` or paused again with `GLMakie.pause_renderloop!(screen)`.
+* `vsync = false`: enables vsync for the window.
+* `framerate = 30.0`: sets the currently rendered frames per second.
+
+## GLFW window attributes
+* `float = false`: Lets the opened window float above anything else.
+* `focus_on_show = false`: Focusses the window when newly opened.
+* `decorated = true`: shows the window decorations or not.
+* `title::String = "Makie"`: Sets the window title.
+* `fullscreen = false`: Starts the window in fullscreen.
+* `debugging = false`: Starts the GLFW.Window/OpenGL context with debug output.
+* `monitor::Union{Nothing, GLFW.Monitor} = nothing`: Sets the monitor on which the Window should be opened.
+
+## Preproccessor
+* `oit = true`: Enles order independent transparency for the window.
+* `fxaa = true`: Enables fxaa (anti-aliasing) for the window.
+* `ssao = true`: Enables screen space occlusion, which gives 3D meshes a soft shadow towards their edges.
```

Comment:
We have ssao off by default on master because a bunch of people had issues with the extra vram it uses

Reply:
Also shouldn't these be post-processors since they run after standard rendering?

Reply:
and it's screen space _ambient_ occlusion, which simulates natural shadowing at inner edges and crevices

Reply:
towards the edges sounds to me like every edge gets like a gradient effect :)